src/web/components/KeyPairs/KeyPairHelper.ts

```diff
@@ -0,0 +1,12 @@
+import { KeyPairModel } from './KeyPairModel';
+
+export const validateUniqueKeyPairName = (
+  value: string,
+  existingKeyPairs: KeyPairModel[] | undefined
+) => {
+  const existingKeyPairNames = existingKeyPairs?.map((keypair) => keypair.name);
```

Comment:
nit: you can probably simplify this to something like (existingKeyPairs.filter((k) => k.name === value).length > 0)